connectivity/check/test.go

```diff
@@ -311,7 +311,7 @@ func (t *Test) Run(ctx context.Context, index int) error {
 	// in which case this function executes as normal.
 	defer func() {
 		// Run all the Test's registered finalizers.
-		t.finalize()
+		t.finalize(ctx)
```

Comment:
We may not want to propagate the same test context to finalizers, as it may already have been canceled at that point. Consider for instance a long running test, which gets interrupted by the user via ^C. At that point, the context will be canceled, and possible finalizers would not be executed, possibly leaving dangling resources. It is probably better to use a separate context, associated with a tight timeout and/or wired again to the interrupts or. WDYT?

Reply:
Hmm, good call. Maybe for now we could pass `context.TODO()` (as it's done in https://github.com/cilium/cilium-cli/blob/main/connectivity/check/policy.go#L531-L534)?

Reply:
Please see the 2nd commit.

Reply:
Looks good to me, thanks!